### PR TITLE
D3-265 | Feature white list

### PR DESCRIPTION
**Pull request by Pobepto**
# Description
Add white list for the front end part. When user try to withdraw money it should select address from list of available addresses.
# Screenshots
**Wallets page -> Withdraw modal**
image
**Settings page**
image

#### Works done
- [x] Added getter for white list
- [x] Changed input field to select field
- [x] Added privacy list to settings page
- [x] Changed script to send key 'eth_whitelist' to account via `setAccountDetail`

# How to check
1. `yarn` to install new packages.
2. `yarn serve` to run dev server.

### CODE REVIEW

File: src/store/Account.js [hunk 1]
```diff
@@ -102,6 +102,11 @@ const getters = {
   ethWalletAddress (state) {
     let wallet = find('ethereum_wallet', state.accountInfo)
     return wallet ? wallet.ethereum_wallet : 'no eth deposit address'
+  },
+
+  withdrawWalletAddresses (state) {
+    const wallet = find('eth_whitelist', state.accountInfo)
+    return wallet ? [wallet.eth_whitelist] : []
```
Comment (oimou) [inline src/store/Account.js]: It seems that only one address can be set, but I suspect that multiple addresses must be required. As far as I see, Notary verifies whether an address is whitelisted at [this line](https://github.com/d3ledger/notary/blob/f32e9887ca3cd5da3556c9110d29cd765d21680a/src/main/kotlin/notary/endpoint/eth/EthRefundStrategyImpl.kt#L143) with splitting eth_whitelist with `,`. So maybe we should split it as well?

**Comment by Pobepto** [comment]
Now withdraw service can validate both ETH and BTC wallets addresses. This is a part of D3-288 task

File: scripts/setup-accounts-and-assets.js [hunk 2]
```diff
@@ -31,15 +30,15 @@ console.log(`accounts: ${accounts.join(', ')}`)
 console.log(`assets: ${wallets.map(w => w.name).join(', ')}`)
 console.log('')
 
-irohaUtil.login(`${testAccFull}`, testPrivKeyHex, nodeIp)
+irohaUtil.login(testAccFull, testPrivKeyHex, nodeIp)
   .then(() => tryToCreateAccount(aliceAccName, irohaDomain, alicePubKey))
   .then(() => irohaUtil.setAccountDetail(testPrivKeyHex, testAccFull, 'ethereum_wallet', '0xAdmin-ethereum_wallet'))
+  .then(() => irohaUtil.setAccountDetail(testPrivKeyHex, testAccFull, 'eth_whitelist', '0xWhitelist-ethereum_wallet'))
```
Comment (oimou) [inline scripts/setup-accounts-and-assets.js]: I think that it should be updated to match the validation rule :)

File: src/components/mixins/inputValidation.js [hunk 3]
```diff
@@ -19,8 +19,7 @@ const set = {
     { pattern: /^([a-z0-9\-.]*)\.(([a-z]{2,4})|([0-9]{1,3}\.([0-9]{1,3})\.([0-9]{1,3})))|(:[0-9]{1,5})$/, message: 'Invalid IP', trigger: 'change' }
   ],
   walletAddress: [
-    { required: true, message: 'Please input wallet address', trigger: 'change' },
-    { pattern: /^0x[a-fA-F0-9]{40}$/, message: 'Invalid wallet address', trigger: 'change' }
```
Comment (oimou) [inline src/components/mixins/inputValidation.js]: I'm not sure but doesn't it work if add both BTC and ETH patterns here? 🤔

Comment (Pobepto) [reply src/components/mixins/inputValidation.js]: I'm trying to do something like this:
``` javascript
  walletAddress: [
    { required: true, message: 'Please input wallet address', trigger: 'change' },
    { pattern: /^[13][a-km-zA-HJ-NP-Z1-9]{25,34}$/, message: 'Invalid wallet address', trigger: 'change' },
    { pattern: /^0x[a-fA-F0-9]{40}$/, message: 'Invalid wallet address', trigger: 'change' }
  ]
```
but this don't work. It always says that wallet address is invalid.

**Pull request merged**